src/arithmetic/list_funcs/list_funcs.c

```diff
@@ -105,6 +108,96 @@ SIValue AR_TOLIST(SIValue *argv, int argc, void *private_data) {
 	return array;
 }
 
+/* Convert a list of values to a list of boolean values.
+   The conversion of each item in list is done using toBooleanOrNull */
+SIValue AR_TOBOOLEANLIST(SIValue *argv, int argc, void *private_data) {
+	ASSERT(argc == 1);
+	if(SI_TYPE(argv[0]) == T_NULL) {
+		return SI_NullVal();
+	}
+	ASSERT(SI_TYPE(argv[0]) == T_ARRAY);
+	SIValue originalArray = argv[0];
+
+	// get array length
+	uint32_t arrayLen = SIArray_Length(originalArray);
+
+	SIValue array = SI_Array(arrayLen);
+	for(uint i = 0; i < arrayLen; i++) {
+		SIValue v = SIArray_Get(originalArray, i);
+		SIArray_Append(&array, AR_TO_BOOLEAN(&v, 1, NULL));
+		SIValue_Free(v);
+	}
+	return array;
+}
+
+/* Convert a list of values to a list of float values.
+   The conversion of each item in list is done using toFloatOrNull */
+SIValue AR_TOFLOATLIST(SIValue *argv, int argc, void *private_data) {
```

Comment:
Generally speaking all 3 functions:
`AR_TOBOOLEANLIST`, `AR_TOFLOATLIST` and `AR_TOINTEGERLIST` are doing practically the same,
consider introducing a static local function which handles all 3 cases e.g.
```c
static SIValue _AR_TOLIST
(
    SIValue *list,                                         // list of elements to convert
    SIValue (*convertor_ptr)(SIValue)) // convert function e.g. AR_TOFLOAT
) {
...
}
```